tests/lang/s05_pragmas/s02_misc/t06_object.nim

```diff
@@ -0,0 +1,24 @@
+block union_pragma:
+  ## The union pragma can be applied to any object type. It means all of the object's 
+  ## fields are overlaid in memory.This produces a union instead of a struct in the 
+  ## generated C/C++ code.
+  type
+    Union {.union.} = object
+      field1: int
+      field2: array[12, int]
+      field3: tuple[idx1, idx2: int]
+
+  doAssert sizeof(Union) == max(sizeof(int), sizeof(array[12, int]))
```

Comment:
So we prove the size works in earlier tests?

Likely need to more todos about this for backend/platform spec tests.

Reply:
We should max all three